### PR TITLE
Added ANTsSerializer to replace previous `_int_antsProcessArguments` and `get_pointer_string`

### PR DESCRIPTION
**Pull request by dipterix**
Added `ANTsSerializer` in `process_args.py` to fix https://github.com/ANTsX/ANTsPy/issues/429 on Windows

It handles image pointers differently on windows: ANTsImages are saved to temporary file instead of returning memory pointers. However, the temporary files need to be cleaned after running command.

This class requires `with` syntax:

```py
with utils.ANTsSerializer() as serializer:
	out = serializer.get_pointer_string(image)
	...
	params = int_antsProcessArguments(args)
	libfn(params)
```

I can't compile ANTsPy from source on my machine (neither osx nor windows). Calling for help if anyone can test the code : )

### CODE REVIEW

**Comment by dipterix** [comment]
@stnava @tfmoraes It'll be great if you could help, also feel free to edit PR if errors out, or to close the PR if you don't like the fix : )

File: ants/utils/process_args.py [hunk 1]
```diff
@@ -20,6 +24,24 @@
...
+def temp_dir():
+    global temporary_folder
+    if temporary_folder is None or not os.path.exists(temporary_folder):
+        temporary_folder = tempfile.mkdtemp(prefix="ANTsPyTmp")
+    return temporary_folder
+
+def temp_file(suffix=None, prefix=None, dir=None):
+    if dir is None:
+        dir = temp_dir()
+    return tempfile.mktemp(suffix=suffix, prefix=prefix, dir=dir)
```
Comment (cookpa) [inline ants/utils/process_args.py:38]: tempfile.mktemp is deprecated and may lead to naming collisions

https://docs.python.org/3/library/tempfile.html#deprecated-functions-and-variables

Comment (dipterix) [reply ants/utils/process_args.py:38]: Will fix this week

**Comment by cookpa** [comment]
Another Windows user posted an issue (#437) so I did some digging. I think the issue might be that the serialization on windows produces "0[address]", whereas on other platforms it is "0x[address]", where "address" is 16 hex digits for a 64-bit address (it would be good to not hard code this requirement!)

There's two possible solutions

1. Modify the proposed serializer to reformat the pointers. This assumes that an address having string form "0x[address]" would cast correctly to a pointer on Windows.

2. Modify ANTs C++  such that "0[address]" is treated as a pointer, much as "0x[address]" is - I think I can do this.

In both cases, I would need someone to test on Windows.

**Comment by stnava** [comment]
Note: we use this same trick in ANTsR - so I would prefer the 2nd approach if it is possible.

**Pull request closed without merging**